lib/SILOptimizer/Mandatory/MoveOnlyAddressCheckerUtils.cpp

```diff
@@ -631,9 +631,17 @@ void swift::siloptimizer::searchForCandidateAddressMarkMustChecks(
 
 namespace {
 
+using ConsumingBlocksCollection = SmallPtrSetVector<SILBasicBlock *, 8>;
+using ConsumingBlocks =
+    llvm::SmallMapVector<TypeTreeLeafTypeRange, ConsumingBlocksCollection, 8>;
```

Comment:
Can this be a single `SmallMapVector<SILBasicBlock*, SmallBitSet>` to map each ConsumingBlock to the fields that are consumed? When you query a field's consumes, you would just filter the result.

That would be consistent with how we represent per-block liveness state.
There are an exponential number of ranges. Whereas blocks are linear, and in practice we have a very small number, and field assignments often occur in the same block.

Reply:
Switched over to this storage.